daemon/src/db/cull_dlcs.rs

```diff
@@ -0,0 +1,187 @@
+use crate::db::Connection;
+use anyhow::Result;
+use model::EventKind;
+
+impl Connection {
+    #[allow(dead_code)]
+    async fn cull_old_dlcs(&self) -> Result<()> {
+        let mut conn = self.inner.acquire().await?;
+
+        let rows = sqlx::query!(
+            r#"
+SELECT
+    id,
+    name,
+    data
+FROM
+    events
+WHERE
+    name = $1 OR
+    name = $2
+ORDER BY
+    created_at DESC
+LIMIT -1 OFFSET 1
+            "#,
+            model::EventKind::CONTRACT_SETUP_COMPLETED_EVENT,
+            model::EventKind::ROLLOVER_COMPLETED_EVENT,
+        )
+        .fetch_all(&mut conn)
+        .await?;
+
+        for row in rows.into_iter() {
+            let id = row.id;
+            let data = EventKind::from_json(row.name, row.data)?;
+
+            let event = match data {
+                EventKind::ContractSetupCompleted { dlc: Some(_), .. } => {
+                    EventKind::ContractSetupCompleted { dlc: None }
+                }
+                EventKind::RolloverCompleted {
+                    dlc: Some(_),
+                    funding_fee,
+                } => EventKind::RolloverCompleted {
+                    dlc: None,
+                    funding_fee,
+                },
+                _ => continue,
+            };
+            let (_, data) = event.to_json();
+
+            let query_result = sqlx::query!(
+                r#"
+UPDATE
+    events
+SET
+    data = $1
+WHERE
+    id = $2
+            "#,
+                data,
+                id,
+            )
+            .execute(&mut conn)
+            .await?;
+
+            if query_result.rows_affected() != 1 {
+                anyhow::bail!("Failed to overwrite old DLC");
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::db::load_cfd_events;
+    use crate::db::memory;
+    use bdk::bitcoin::Amount;
+    use model::Cfd;
+    use model::CfdEvent;
+    use model::EventKind;
+    use model::FundingRate;
+    use model::Leverage;
+    use model::OpeningFee;
+    use model::OrderId;
+    use model::Position;
+    use model::Price;
+    use model::Role;
+    use model::Timestamp;
+    use model::TxFeeRate;
+    use model::Usd;
+    use rust_decimal_macros::dec;
+    use sqlx::Connection;
+    use time::Duration;
+
+    #[tokio::test]
+    async fn given_two_rollovers_when_cull_old_dlcs_only_last_one_has_some_dlc() {
```

Comment:
🙃 I would add another test for the case where we have only one rollover and cull to ensure that this rollover is not removed.

Additionally I'd add one test with three rollovers to ensure that we remove the first two rollovers - but this one is not as important.

Reply:
Yeah, I will add a few more tests!

I just wanted to prove that it seems to work before continuing.